vowpalwabbit/cb_sample.cc

```diff
@@ -58,8 +59,8 @@ struct cb_sample_data
         if (strncmp(examples[0]->tag.begin(), SEED_IDENTIFIER.c_str(), SEED_IDENTIFIER.size()) == 0 &&
             examples[0]->tag.size() > SEED_IDENTIFIER.size())
         {
-          substring tag_seed{examples[0]->tag.begin() + 5, examples[0]->tag.begin() + examples[0]->tag.size()};
-          seed = uniform_hash(tag_seed.begin, substring_len(tag_seed), 0);
+          VW::string_view tag_seed(examples[0]->tag.begin() + 5, examples[0]->tag.size());
```

Comment:
No, string_view is just a `const char*` (to an existing string) and length. The consequence of this is that the underlying string needs to outlive the string_view since it owns no memory